TGEngine/drawlib/Beziercurve.cpp

```diff
@@ -1,5 +1,37 @@
 #include "Beziercurve.hpp"
 
-void drawBeziercurve(TGVertex ver1, TGVertex ver2, TGVertex ver3, int max_verticies, VertexBuffer* verticeis) {
-	
+static TGVertex midpoint(TGVertex a, TGVertex b, float t)
+{
+    return {
+        b.position + (a.position - b.position) * t,
+        math::mixColor(a.color, b.color, t),
+        b.uv + (a.ub - b.uv) * t,
+        COLOR_ONLY
+    };
+}
+
+void drawBeziercurve(VertexBuffer* vertices, IndexBuffer* indices, size_t max_vertices, vector<TGVertex>* points) {
+    if(points->size() == 0) return;
+    if(points->size() == 1)
+    {
+        vertices->add(points->at(0));
+        return;
+    }
+    if(points->size() == 2)
+    {
+        for(float i = 0; i < max_vertices; i++)
+            vertices->add(midpoint(points->at(0), points->at(1), i / (max_vertices - 1));
+        return;
+    }
+    for(float t = 0; t < max_vertices; t++)
```

Comment:
A render needs to consist out of triangles!
Means you need to triangulate everything with a triangle fan algorithm.
For a convex polygon you get  `max_vertices - 2` triangles -> so `(max_vertices - 2) * 3` Vertices